src/fabric/src/fabric_rpc.erl

```diff
@@ -638,16 +638,14 @@ make_att_readers([#doc{atts = Atts0} = Doc | Rest]) ->
     [Doc#doc{atts = Atts} | make_att_readers(Rest)].
 
 make_att_reader({follows, Parser, Ref}) ->
+    % This code will fail if the returned closure is called by a
+    % process other than the one that called make_att_reader/1 in the
+    % first place. The reason we don't put everything inside the
+    % closure is that the `hello_from_writer` message must *always* be
+    % sent to the parser, even if the closure never gets called.
+    ParserRef = erlang:monitor(process, Parser),
+    Parser ! {hello_from_writer, Ref, self()},
```

Comment:
I had noticed we previously cached the PRef in the `mp_parser_ref` process dict. So, for multiple attachments we would have created only a single monitor.

In the PR we have a separate monitor for each attachment and send a `hello_from_writer` for each one.  If it's just saving an extra few monitor references it's not a big deal (though they are remote, not sure actually how impactful those are...). But would multiple `hello_from_writer` affect the by fetching protocol correctness. Each one would end up overwriting the `WriterPid => {WriterRef, 0}` entry but monitors would stack up, then we could get multiple `DOWN` messages for each attachment instead of once per writer termination.

Reply:
Ah, I totally missed the 1 parser + multiple attachments line of thinking. I was looking at the caching and figuring it was only there because the function was being invoked for every `get_bytes` call, but yes, what you're proposing definitely makes sense given the lifecycle of the parser. Thanks for pointing it out!

And yes, it makes sense to extend the test suite here.

Reply:
I had tried to handle multiple attachments in a new set of commits. Also rebased the branch on latest 3.x